java/memory/memory-core/src/main/java/org/apache/arrow/memory/DefaultAllocationManagerOption.java

```diff
@@ -114,10 +114,20 @@ static AllocationManagerType getDefaultAllocationManagerType() {
   }
 
   private static AllocationManager.Factory getUnsafeFactory() {
-    return getFactory("org.apache.arrow.memory.UnsafeAllocationManager");
+    try {
+      return getFactory("org.apache.arrow.memory.UnsafeAllocationManager");
+    } catch (RuntimeException e) {
+      throw new RuntimeException("Please add arrow-memory-unsafe to your classpath," +
+          " No DefaultAllocationManager found to instantiate an UnsafeAllocationManager", e);
+    }
   }
 
   private static AllocationManager.Factory getNettyFactory() {
-    return getFactory("org.apache.arrow.memory.NettyAllocationManager");
+    try {
+      return getFactory("org.apache.arrow.memory.NettyAllocationManager");
+    } catch (RuntimeException e) {
+      throw new RuntimeException("Please add arrow-memory-unsafe to your classpath," +
```

Comment:
should be `arrow-memory-netty` and `NettyAllocationManager` below

Reply:
sigh....sorry. Copy paste for the win. Fixed.